OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -483,196 +491,104 @@ public virtual void Stop()
 		}
 	}
 
-	class OpenAlStreamingSound : OpenAlSound
+	class OpenAlAsyncLoadSound : OpenAlSound
 	{
-		const int BufferCount = 3;
-		const int BufferSizeInSecs = 1;
-		readonly object bufferDequeueLock = new object();
+		static readonly byte[] SilentData = new byte[2];
 		readonly CancellationTokenSource cts = new CancellationTokenSource();
-		readonly Task streamTask;
-		readonly Stack<uint> freeBuffers = new Stack<uint>(BufferCount);
-		int totalSamplesPlayed;
+		readonly Task playTask;
 
-		public OpenAlStreamingSound(uint source, bool looping, bool relative, WPos pos, float volume, int channels, int sampleBits, int sampleRate, Stream stream)
+		public OpenAlAsyncLoadSound(uint source, bool looping, bool relative, WPos pos, float volume, int channels, int sampleBits, int sampleRate, Stream stream)
 			: base(source, looping, relative, pos, volume, sampleRate)
 		{
-			streamTask = Task.Run(async () =>
-			{
-				var format = OpenAlSoundEngine.MakeALFormat(channels, sampleBits);
-				var bytesPerSample = sampleBits / 8;
+			// Load a silent buffer into the source. Without this,
+			// attempting to change the state (i.e. play/pause) the source fails on some systems.
+			var silentSource = new OpenAlSoundSource(SilentData, channels, sampleBits, sampleRate);
```

Comment:
You missed the `AL10.alSourcei(source, AL10.AL_BUFFER, (int)silentSource.Buffer);` here!
With this addition it now works correctly on my machine.

(github is showing the original comments as outdated, so reposting).

Reply:
A derp is me. Fixed.